src/mappings/rewardsDistributor.ts

```diff
@@ -129,6 +159,90 @@ export async function handleRewardsClaimed(
   } catch {
     logger.error(`ERROR: handleRewardsClaimed`);
   }
+
+  await updateEraRewardClaimed(event);
+}
+
+interface EraRewardData {
+  indexerId: string;
+  delegatorId: string;
+  eraId: string;
+  isCommission: boolean;
+  claimed: boolean;
+  amount?: bigint;
+  createdBlock?: number;
+  createdTimestamp?: Date;
+}
+
+async function updateEraReward(data: EraRewardData): Promise<EraReward> {
+  console.log('updateEraReward', data);
+
+  const id = `${data.indexerId}_${data.delegatorId}_${data.eraId}${
+    data.isCommission ? '_commission' : ''
+  }`;
+
+  let eraReward = await EraReward.get(id);
+
+  if (!eraReward) {
+    eraReward = EraReward.create({
+      id,
+      indexerId: data.indexerId,
+      delegatorId: data.delegatorId,
+      eraId: data.eraId,
+      isIndexer: data.indexerId === data.delegatorId,
+      claimed: data.claimed,
+      amount: data.amount ?? BigInt(0),
+      createdBlock: data.createdBlock,
+      createdTimestamp: data.createdTimestamp ?? new Date(),
+    });
+  } else {
+    eraReward.claimed = data.claimed;
+
+    eraReward.amount = data.amount ?? eraReward.amount;
+    eraReward.createdBlock = data.createdBlock ?? eraReward.createdBlock;
+    eraReward.createdTimestamp =
+      data.createdTimestamp ?? eraReward.createdTimestamp;
+  }
+
+  await eraReward.save();
+  return eraReward;
+}
+
+async function updateEraRewardClaimed(
+  event: EthereumLog<ClaimRewardsEvent['args']>
+): Promise<void> {
+  logger.info('updateEraRewardClaimed');
+  assert(event.args, 'No event args');
+
+  const { indexer, delegator } = event.args;
+  const id = `${indexer}_${delegator}`;
+
+  let eraRewardClaimed = await EraRewardClaimed.get(id);
+  if (!eraRewardClaimed) {
+    eraRewardClaimed = EraRewardClaimed.create({
+      id,
+      lastClaimedEra: 0,
+    });
+  }
+
+  let lastClaimedEra = eraRewardClaimed.lastClaimedEra;
+  const hasNextEraReward = true;
+
+  while (hasNextEraReward) {
+    const eraReward = await EraReward.get(`${id}_${lastClaimedEra + 1}`);
```

Comment:
Need to compare with current era id. only era up to `CurrentEra-1` can be marked as claimed.